cli/azd/internal/appdetect/docker.go

```diff
@@ -70,3 +71,79 @@ func parsePortsInLine(s string) ([]Port, error) {
 	}
 	return ports, nil
 }
+
+func AddDefaultDockerfile(project Project) (*Docker, error) {
+	path := project.Path
+	// if Dockerfile not exists, provide a default one
+	if project.Language == Java {
+		log.Printf("Dockerfile not found, will provide a default one")
+		_, hasParentPom := project.Options[JavaProjectOptionMavenParentPath]
+		err := writeDockerfileIntoFs(path, hasParentPom)
+		if err != nil {
+			return nil, err
+		}
+		dockerfilePath := filepath.Join(path, "Dockerfile")
+		return AnalyzeDocker(dockerfilePath)
+	}
+
+	return nil, nil
+}
+
+const (
+	DockerfileSingleStage = `FROM openjdk:21-jdk-slim
+COPY ./target/*.jar app.jar
+COPY ./target/*.war app.war
+ENTRYPOINT ["sh", "-c", \
+    "if [ -f /app.jar ]; then java -jar /app.jar; \
+    elif [ -f /app.war ]; then java -jar /app.war; \
+    else echo 'No JAR or WAR file found'; fi"]`
+
+	DockerfileMultiStage = `FROM maven:3 AS build
+WORKDIR /app
+COPY . .
+RUN mvn --batch-mode clean package -DskipTests
+
+FROM openjdk:21-jdk-slim
+WORKDIR /
+COPY --from=build /app/target/*.jar app.jar
+COPY --from=build /app/target/*.war app.war
+ENTRYPOINT ["sh", "-c", \
+    "if [ -f /app.jar ]; then java -jar /app.jar; \
+    elif [ -f /app.war ]; then java -jar /app.war; \
+    else echo 'No JAR or WAR file found'; fi"]`
+)
+
+func writeDockerfileIntoFs(path string, hasParentPom bool) error {
+	if _, err := os.Stat(path); err != nil {
+		return fmt.Errorf("error accessing path %s: %w", path, err)
+	}
+
+	dockerfilePath := filepath.Join(path, "Dockerfile")
```

Comment:
I have a concern here that whether we should output this Dockerfile to user's source folder. But it's okay for now, let's keep it the straightforward way before we have a better idea.

Reply:
Here is my suggestion: 
1. Not create docker when app detect. But return the information exists when do app detect for java applications.
2. Create the dockerfile when use it to build docker image.
3. Delete the docker file after docker image build finished.

Reply:
Per offline sync, we will move this logic into `azd package` phrase, and add the default Dockerfile into temp dir, not user source code folder.